core/unix/os.c

```diff
@@ -2698,12 +2705,7 @@ os_should_swap_state(void)
     return (INTERNAL_OPTION(mangle_app_seg) &&
             IF_CLIENT_INTERFACE_ELSE(INTERNAL_OPTION(private_loader), false));
 #elif defined(AARCHXX)
-    /* FIXME i#1582: this should return true, but there is a lot of complexity
-     * getting os_switch_seg_to_context() to do the right then when called
-     * at main thread init, secondary thread init, early and late injection,
-     * and thread exit, since it is fragile with its writes to app TLS.
-     */
-    return false;
+    return IF_CLIENT_INTERFACE_ELSE(INTERNAL_OPTION(private_loader), false);
```

Comment:
@derekbruening 
I found this change to be the source of a regression on Android - an application run under DR (without any extra options or clients) is now failing with "Cannot correctly handle received signal 11 in thread X.".

`INTERNAL_OPTION(private_loader)` here reports `true` on Android, but private loader doesn't really work there (cc https://github.com/DynamoRIO/dynamorio/issues/3543), and in this particular case something breaks if it's considered working.

I guess, since things generally work on Android, considering `INTERNAL_OPTION(private_loader)` true on Android is not *too* harmful, but apparently there's an unknown set of cases where it is indeed harmful.
Not sure what is the best course of action here, besides actually implementing the private loader.

Reply:
Hmm, the real problem is that there is no CI testing for Android: #4639.  The code reflects the state as of several years ago, when the private loader did work.  Like you said, the best solution would be to get it working with the latest Android, though if you have a quick fix that improves things in the short term please submit a PR.

Unfortunately while the handful of regular DR maintainers are happy to provide advice and consulting for Android we are not likely to be able to spend time on actual coding work targeting Android.  Several people have expressed interest in Android support; hopefully someone will contribute some time.

Reply:
@petrochenkov if you could file a new issue on the regression; and ideally submit a PR with a quick fix if it's blocking things.

Reply:
The quick fix that seems to work in practice is to continue returning `false` in the `defined(ANDROID)` case, unfortunately I cannot submit it without an approval.

Reply:
> The quick fix that seems to work in practice is to continue returning false in the defined(ANDROID) case, unfortunately I cannot submit it without an approval.

You mean some approval process from your employer I assume.